src/components/tasks/task-dialog.tsx

```diff
@@ -199,6 +270,152 @@ const TaskDialog = ({ projectId, milestoneId, open, task, onClose }: Props) => {
     setTaskData({ ...taskData, [field]: value });
   };
 
+  /**
+   * Handles change proposal update form date change
+   *
+   * @param field string
+   * @param changeProposalId string
+   * @param value date
+   */
+  const handleUpdateChangeProposalDateFormChange =
+    (field: keyof ChangeProposal, changeProposalId?: string) => (value: DateTime<boolean> | null) => {
+      if (!changeProposalId || !value) return;
+
+      const updatedProposals = updateChangeProposalData.map((proposal) => {
+        if (proposal.id === changeProposalId) {
+          // TODO: This is needed to ensure correct date is set when the initial value is undefined
+          const date = value.toJSDate();
+          date.setHours(3);
```

Comment:
Why do we need the hours to be exactly 3?

Reply:
For some reason when the date state was previously undefined we get a default time returned of 00:00 with a time zone of +3 which results in the incorrect date being stored. I' not sure why this is and when the previous value is not undefined we get a time of 03:00 +3, again not sure why.
Perhaps we could look at this together at some point?